src/uwtools/tests/test_rocoto.py

```diff
@@ -271,3 +294,35 @@ def test__tag_name(self, instance):
         assert instance._tag_name("foo") == ("foo", "")
         assert instance._tag_name("foo_bar") == ("foo", "bar")
         assert instance._tag_name("foo_bar_baz") == ("foo", "bar_baz")
+
+
+# Schema tests
+
+
+def test_schema_workflow_cycledef():
+    errors = validator("workflow", "cycledef")
+    # Basic spec:
+    spec = "202311291200 202312011200 06:00:00"
+    assert not errors([{"spec": spec}])
+    # Spec with step specified as seconds:
+    assert not errors([{"spec": "202311291200 202312011200 3600"}])
+    # Basic spec with group attribute:
+    assert not errors([{"attrs": {"group": "g"}, "spec": spec}])
+    # Spec with positive activation offset attribute:
+    assert not errors([{"attrs": {"activation_offset": "12:00:00"}, "spec": spec}])
+    # Spec with negative activation offset attribute:
+    assert not errors([{"attrs": {"activation_offset": "-12:00:00"}, "spec": spec}])
+    # Spec with activation offset specified as positive seconds:
+    assert not errors([{"attrs": {"activation_offset": "3600"}, "spec": spec}])
+    # Spec with activation offset specified as negative seconds:
+    assert not errors([{"attrs": {"activation_offset": "-3600"}, "spec": spec}])
+    # Property spec is required:
+    assert "'spec' is a required property" in errors([{}])
+    # Additional properties are not allowed:
+    assert "'foo' was unexpected" in errors([{"spec": spec, "foo": "bar"}])
+    # Additional attributes are not allowed:
+    assert "'foo' was unexpected" in errors([{"attrs": {"foo": "bar"}, "spec": spec}])
+    # Bad spec:
+    assert "'x 202312011200 06:00:00' is not valid" in errors([{"spec": "x 202312011200 06:00:00"}])
+    # Spec with bad activation offset attribute:
+    assert "'foo' is not valid" in errors([{"attrs": {"activation_offset": "foo"}, "spec": spec}])
```

Comment:
This is the first of what should probably be many tests of the UW YAML schemas. It works by extracting from the overall schema a sub-schema corresponding to the small bit of YAML to be tested, via the positional arguments to the `validator()` helper function, which define directions to the sub-schema of interest. `validator()` returns a function (bound to local variable `errors` here) that can be called with a block of UW YAML to test against that sub-schema, and which returns a string of validation errors (empty if no errors were encountered), which can be tested for the presence of expected error messages.

Hopefully this mechanism will support adding more schema tests so we can directly test fine-grained behaviors of our schemas.